src/cpp/shared/combat/army.cpp

```diff
@@ -614,6 +650,293 @@ void army::Walk(signed int dir, int last, int notFirst) {
+    DelayTil(&glTimers);
+  }
+  
+  // occupy new hex
+  this->occupiedHex = toHex;
+  gpCombatManager->combatGrid[this->occupiedHex].unitOwner = LOBYTE(this->owningSide);
+  gpCombatManager->combatGrid[this->occupiedHex].stackIdx = LOBYTE(this->stackIdx);
+  gpCombatManager->combatGrid[this->occupiedHex].occupiersOtherHexIsToLeft = -1;
+
+  // reset renderer settings
+  savedscreen->CopyTo(gpWindowManager->screenBuffer, 0, 0, 0, 0, INTERNAL_WINDOW_WIDTH, INTERNAL_WINDOW_HEIGHT); // clear the screen from the previous creature sprite
+  giMinExtentY = giMinExtentX = giMaxExtentY = giMaxExtentX = 0;
+  // reverting frame
+  this->frameInfo.animationFrameToImgIdx[ANIMATION_TYPE_MOVE][0] = 1;
+
+  delete savedscreen;
+}
+
+extern searchArray *gpSearchArray;
+
+int army::WalkTo(int hex) {
+  this->targetStackIdx = -1;
+  this->targetOwner = this->targetStackIdx;
+  if (gpCombatManager->hasMoat && this->creature.creature_flags & TWO_HEXER) {
+    bool goingToMoat = false;
+    int moatIdx = 0;
+    for (int i = 0; i < 9; ++i) {
+      if (moatCell[i] == hex) {
+        goingToMoat = true;
+        moatIdx = i;
+      }
+    }
+    if (goingToMoat) {
+      bool onEnemySideOfMoat = false;
+      if (moatIdx == 4 && gpCombatManager->drawBridgePosition != 4)
+        onEnemySideOfMoat = true;
+      if (moatIdx > 0 && *(&giWalkingYMod + moatIdx + 3) == this->occupiedHex// moatCell[moatIdx-1]
+        || moatIdx < 8 && moatCell[moatIdx + 1] == this->occupiedHex)
+        onEnemySideOfMoat = true;
+      for (int j = 0; j < 6; ++j) {
+        if (moatCell[moatIdx] == army::GetAdjacentCellIndex(this->occupiedHex, j))
+          onEnemySideOfMoat = true;
+      }
+      if (!this->owningSide && moatCell[this->occupiedHex / 13] < this->occupiedHex)
+        onEnemySideOfMoat = true;
+      if (this->owningSide == 1 && moatCell[this->occupiedHex / 13] > this->occupiedHex)
+        onEnemySideOfMoat = true;
+      if (!onEnemySideOfMoat) {
+        if (this->facingRight == 1)
+          --hex;
+        else
+          ++hex;
+      }
+    }
+  }
+
+  if(this->FindPath(this->occupiedHex, hex, this->creature.speed, 1, 0)) {
+    int traveledHexes = 0;
+    int initialHex = this->occupiedHex;
+    for(int hexIdxb = gpSearchArray->field_8 - 1; hexIdxb >= 0; --hexIdxb) {
+      int dir = *((BYTE *)&gpSearchArray->field_2418 + hexIdxb);
+      int destHex = this->GetAdjacentCellIndex(this->occupiedHex, dir);
+      if(this->creatureIdx == CREATURE_CYBER_PLASMA_BERSERKER && gIronfistExtra.combat.stack.abilityCounter[this]) {
+        if(gMoveAttack && hexIdxb < 4) { // less than 4 hexes from enemy
+          // find last hex
+          for(int h = hexIdxb; h >= 0; --h) {
+            dir = *((BYTE *)&gpSearchArray->field_2418 + h);
+            destHex = this->GetAdjacentCellIndex(this->occupiedHex, dir);
+            this->occupiedHex = destHex;
+          }
+          this->ArcJump(initialHex, destHex);
```

Comment:
Is this the main thing that changed here?

Reply:
The code goes to return 0 here because we don't want standing animation to show

Reply:
Can you use an API like this? http://liballeg.org/stabledocs/en/alleg013.html#do_arc

Reply:
Oh right! I can make an array of coordinates using a function (I guess you replied in the wrong place)

Reply:
Doing an array also works. The suggestion is to have a callback, and invoke the callback once for each point in the arc.

Get familiar with refactoring code to use callbacks if you're not already. https://www.joelonsoftware.com/2006/08/01/can-your-programming-language-do-this/

Reply:
Oh....yes, I did respond in the wrong place.

Reply:
I don't think it's easier to make and use and understand a function that calculates a new point from a previous point of an arc, than to calculate all the points in the arc at once and return them as an array. Not sure if I understood your idea of "callbacks" correctly..

Reply:
No, that's not what I suggested. Look at do_arc from Allegro and understand how it's used.